src/main/diagnostics/rdBinInShell.ts

```diff
@@ -18,43 +19,45 @@ mainEvents.on('settings-update', (cfg) => {
 class RDBinInShellPath implements DiagnosticsChecker {
   constructor(id: string, executable: string, ...args: string[]) {
     this.id = id;
-    this.executable = executable;
+    if (['darwin', 'linux'].includes(os.platform())) {
+      this.executable = which.sync(executable, { nothrow: true }) ?? '';
+    }
     this.args = args;
   }
 
   id: string;
-  executable: string;
+  executable = '';
   args: string[];
   category = DiagnosticsCategory.Utilities;
-  async applicable(): Promise<boolean> {
-    if (!['darwin', 'linux'].includes(os.platform())) {
-      return false;
-    }
-    try {
-      await fs.promises.access(this.executable, fs.constants.X_OK);
-
-      return true;
-    } catch (ex) {
-      return false;
-    }
+  applicable(): Promise<boolean> {
+    return Promise.resolve(!!this.executable);
   }
 
   async check() {
-    const { stdout } = await spawnFile(this.executable, this.args, { stdio: 'pipe' });
-    const dirs = stdout.trim().split(':');
-    const desiredDirs = dirs.filter(p => p === paths.integration);
-    const passed = desiredDirs.length > 0;
     const fixes: {description: string}[] = [];
-    const exe = path.basename(this.executable);
-    let description = `The ~/.rd/bin directory has not been added to the PATH, so command-line utilities are not configured in your ${ exe } shell.`;
+    let passed = false;
+    let description = '';
```

Comment:
```suggestion
    let description: string;
```
Let the TypeScript compiler ensure that we always assign to this before reading it.